### PR TITLE
Support for Bio Multi-protocol keys

### PR DESCRIPTION
**Pull request by jensutbult**

### CODE REVIEW

File: YubiKit/YubiKit/Connections/Shared/Sessions/MGMT/YKFManagementSession.h [hunk 1]
```diff
@@ -47,6 +47,14 @@ typedef void (^YKFManagementSessionGetDeviceInfoBlock)
 ///    parameter is nil.
 typedef void (^YKFManagementSessionWriteCompletionBlock) (NSError* _Nullable error);
 
+/// @abstract
+///    Response block for [deviceReset:completion:] which will do a device reset on a YubiKey Biuo
```
Comment (AdamVe) [inline YubiKit/YubiKit/Connections/Shared/Sessions/MGMT/YKFManagementSession.h:51]: Biuo -> Bio

File: YubiKit/YubiKit/Connections/Shared/Errors/YKFInvalidPinError.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//
+//  YKFInvalidPinError.h
+//  YubiKit
+//
+//  Created by Jens Utbult on 2024-06-19.
+//  Copyright © 2024 Yubico. All rights reserved.
+//
```
Comment (AdamVe) [inline YubiKit/YubiKit/Connections/Shared/Errors/YKFInvalidPinError.h:7]: I saw in other files you update this to a different header, maybe you forgot here?

File: YubiKit/YubiKit/Connections/Shared/Errors/YKFInvalidPinError.m [hunk 3]
```diff
@@ -0,0 +1,29 @@
+//
+//  YKFInvalidPinError.m
+//  YubiKit
+//
+//  Created by Jens Utbult on 2024-06-19.
+//  Copyright © 2024 Yubico. All rights reserved.
+//
```
Comment (AdamVe) [inline YubiKit/YubiKit/Connections/Shared/Errors/YKFInvalidPinError.m:7]: This header is different than in other files.

File: YubiKit/YubiKit/Connections/Shared/Sessions/PIV/YKFPIVSession.m [hunk 4]
```diff
@@ -857,6 +881,91 @@ - (void)blockPuk:(int)counter completion:(YKFPIVSessionGenericCompletionBlock)co
...
+                return;
+            }
+            int retries = [self getRetriesFromStatusCode:(int)error.code];
+            if (retries >= 0) {
+                completion([YKFInvalidPinError invalidPinErrorWithRetries:retries]);
+                return;
+            } else {
+                completion(error);
+                return;
+            }
```
Comment (AdamVe) [inline YubiKit/YubiKit/Connections/Shared/Sessions/PIV/YKFPIVSession.m:964]: Review this part - if verification of the temporary PIN fails, the error.code will be 0x63C0, so there is no need to get the retries.

**Review by AdamVe — CHANGES_REQUESTED** [review]
I found some code snippets to review. 🕵🏻‍♂️

**Review by AdamVe — APPROVED** [review]
LGTM

**Review by dainnilsson — APPROVED** [review]

**Pull request merged**